packages/button/theme/lumo/vaadin-button-styles.js

```diff
@@ -0,0 +1,302 @@
+      }
+
+      :host([theme~='primary']:not([active]):hover)::before {
+        opacity: 0;
+      }
+    }
+
+    :host([theme~='primary'][active])::after {
+      opacity: 0.2;
+    }
+
+    /* Colors (success, error, contrast) */
+
+    :host([theme~='success']) {
+      color: var(--lumo-success-text-color);
+    }
+
+    :host([theme~='success'][theme~='primary']) {
+      background-color: var(--lumo-success-color);
+      color: var(--lumo-success-contrast-color);
+    }
+
+    :host([theme~='success'][theme~='primary'][disabled]) {
+      background-color: var(--lumo-success-color-50pct);
+    }
+
+    :host([theme~='error']) {
+      color: var(--lumo-error-text-color);
+    }
+
+    :host([theme~='error'][theme~='primary']) {
+      background-color: var(--lumo-error-color);
+      color: var(--lumo-error-contrast-color);
+    }
+
+    :host([theme~='error'][theme~='primary'][disabled]) {
+      background-color: var(--lumo-error-color-50pct);
+    }
+
+    :host([theme~='contrast']) {
+      color: var(--lumo-contrast);
+    }
+
+    :host([theme~='contrast'][theme~='primary']) {
+      background-color: var(--lumo-contrast);
+      color: var(--lumo-base-color);
+    }
+
+    :host([theme~='contrast'][theme~='primary'][disabled]) {
+      background-color: var(--lumo-contrast-50pct);
+    }
+
+    /* Icons */
+
+    [part] ::slotted(vaadin-icon),
+    [part] ::slotted(iron-icon) {
+      display: inline-block;
+      width: var(--lumo-icon-size-m);
+      height: var(--lumo-icon-size-m);
+    }
+
+    /* Vaadin icons are based on a 16x16 grid (unlike Lumo and Material icons with 24x24), so they look too big by default */
+    [part] ::slotted(vaadin-icon[icon^='vaadin:']),
+    [part] ::slotted(iron-icon[icon^='vaadin:']) {
+      padding: 0.25em;
+      box-sizing: border-box !important;
+    }
+
+    [part='prefix'] {
+      margin-left: -0.25em;
+      margin-right: 0.25em;
+    }
+
+    [part='suffix'] {
+      margin-left: 0.25em;
+      margin-right: -0.25em;
+    }
+
+    /* Icon-only */
+
+    :host([theme~='icon']:not([theme~='tertiary-inline'])) {
+      min-width: var(--lumo-button-size);
+    }
+
+    :host([theme~='icon']:not([theme~='tertiary-inline'])) [part='button'] {
+      padding-left: calc(var(--lumo-button-size) / 4);
+      padding-right: calc(var(--lumo-button-size) / 4);
+    }
+
+    :host([theme~='icon']) [part='prefix'],
+    :host([theme~='icon']) [part='suffix'] {
+      margin-left: 0;
+      margin-right: 0;
+    }
+
+    /* RTL specific styles */
+
+    :host([dir='rtl']) [part='prefix'] {
+      margin-left: 0.25em;
+      margin-right: -0.25em;
+    }
+
+    :host([dir='rtl']) [part='suffix'] {
+      margin-left: -0.25em;
+      margin-right: 0.25em;
+    }
+
+    :host([dir='rtl'][theme~='icon']) [part='prefix'],
+    :host([dir='rtl'][theme~='icon']) [part='suffix'] {
+      margin-left: 0;
+      margin-right: 0;
+    }
+  `,
+  { moduleId: 'lumo-button' }
```

Comment:
Related to the above comment: let's use `lumo-button-styles` for now (see #2323)

Reply:
Done.